sys-apps/debianutils: drop installkernel use flag

AndrewAmmerlaan opened the pull request:
Random packages requiring some tool from Debian should not cause the kernel installation process to change.

The dropping of the debianutils dependency in ca-certificates has already caused some surprises due to installkernel being depcleaned. The origin of the problem lies here in debianutils, users unknowingly use and rely on `installkernel` but do not have it in their world file because it was implicitly pulled in by some package that happens to use the `run-parts` command.

And also the other way around. If I am one of the users that wants to do everything manually, I should not have my `make install` unknowingly altered by some package that I installed which pulled debianutils into the depgraph.

Drop this unused runtime dependency (which is against policy to begin with) and its accompanying flag. 
This will be accompanied with the following news item:

```
Title: installkernel is no longer implicitly installed
Author: Andrew Ammerlaan <andrewammerlaan@gentoo.org>
Posted: 2024-02-26
Revision: 1
News-Item-Format: 2.0
Display-If-Installed: sys-kernel/installkernel
Display-If-Installed: >=sys-apps/debianutils-5.14-r1
Display-If-Installed: app-misc/ca-certificates

/sbin/installkernel is a script called by the kernel's "make install"
as well as by the distribution kernel's post-install phase. If you are
reading this then chances are you use and rely on installkernel[1] and
what follows is essential for you.

Previously sys-kernel/installkernel was implicitly installed on many
systems via a dependency in sys-apps/debianutils. This dependency was
toggled by the "installkernel" USE flag, and enabled by default.

Until recently, sys-apps/debianutils was in turn pulled in by
app-misc/ca-certificates, an essential package installed on many
systems. However, this dependency of app-misc/ca-certificates on 
sys-apps/debianutils was removed[2]. As a result many users may find
that sys-apps/debianutils and therefore sys-kernel/installkernel are no
longer part of the dependency graph and will therefore be cleaned up by
"emerge --depclean".

Removing sys-kernel/installkernel from your system WILL change the way
kernels are installed by "make install"! Instead of the versioned
/boot/vmlinuz-x.y.z that you are used to, "make install" will simply
copy bzImage (or equivalent for your arch) into /boot. This image may
not be picked up by your bootloader or its configuration tools.

To avoid surprises from such implicit dependencies from happening again
in the future, the dependency on sys-kernel/installkernel in
sys-apps/debianutils is removed. And as such, sys-kernel/installkernel
is only installed on the system if it is either explicitly selected or
pulled in via the distribution kernels (e.g. gentoo-kernel(-bin)).


User Action Required (all users)
====================

Users who currently have sys-kernel/installkernel installed, must
ensure that it is explicitly selected by emerging it:

	emerge --noreplace sys-kernel/installkernel

Users who find that sys-kernel/installkernel has already been cleaned
from their systems and are therefore affected by the change in kernel
installation described above should re-install sys-kernel/installkernel
and then re-install their kernel.

	emerge sys-kernel/installkernel
	cd /usr/src/linux       # (or other location of the kernel sources)
	make install

Note that this re-installation is not required for users of the
distribution kernels (e.g. gentoo-kernel(-bin)).


[1] https://wiki.gentoo.org/wiki/Installkernel
[2] https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=6e6ccafd58bc7401fa371d2f255d72ddae0131e6
````

thesamesam commented:
cc @eli-schwartz

floppym reviewed (approved):
This seems fine to me.

Perhaps the handbook should be updated to advise users to install sys-kernel/installkernel before running `make install` when building a kernel manually.

https://wiki.gentoo.org/wiki/Handbook:AMD64/Installation/Kernel#Alternative:_Manual_configuration

eli-schwartz commented:
+1

eli-schwartz commented:
Thanks for writing the news article too. I was, obviously, blindsided by this side effect of "surely we can just remove a trivial debianutils package that does nothing, right?"

AndrewAmmerlaan commented:
> Perhaps the handbook should be updated to advise users to install sys-kernel/installkernel before running `make install` when building a kernel manually.
> 
> https://wiki.gentoo.org/wiki/Handbook:AMD64/Installation/Kernel#Compiling_and_installing

Done, this was something that needed changing regardless of the change here because it seemed to be implying that `make install` does the copying and installing of the config and System Map, when in fact it does not. Thanks for noticing!

kuraga commented:
> However, this 
dependency was recently removed.

Let's point https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=6e6ccafd58bc7401fa371d2f255d72ddae0131e6 here?

thesamesam reviewed (approved):
OK with the wiki and such updated. Give @SoapGentoo a chance to comment though.

AndrewAmmerlaan commented:
> > However, this
> > dependency was recently removed.
> 
> Let's point https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=6e6ccafd58bc7401fa371d2f255d72ddae0131e6 here?

Added. And also addressed comments from the mailing list.

kuraga commented:
> Note that this re-installation is not required for users of the distribution kernels (e.g. gentoo-kernel(-bin)). 

I'd also put this on the top of the User actions list :)

AndrewAmmerlaan commented:
> > Note that this re-installation is not required for users of the distribution kernels (e.g. gentoo-kernel(-bin)).
> 
> I'd also put this on the top of the User actions list :)

I put it at the bottom because users of the dist-kernels still benefit from having `sys-kernel/installkernel` in their world file even though it is pulled in by the eclass. If they would for example switch to a manual build, then they might be surprised to find that installkernel is depcleaned and that the manually built kernel is not correctly installed.

SoapGentoo reviewed (approved):
agreed, let's get this in

AndrewAmmerlaan commented:
Thanks, merged this and pushed the new news item.
